aio/src/app/custom-elements/events/events.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, OnInit } from '@angular/core';
+
+import { EventsService } from './events.service';
+
+export interface Event {
+  endDate: string;      // "2020-05-27",
+  address: string;      // "Oslo, Norway",
+  title: string;        // "ngVikings",
+  linkUrl: string;      // "https://ngvikings.org/",
+  name: string;          // "ngVikings",
+  info: string;          // "May 25-26 conference, 27 workshops, 2020"
+}
+
+@Component({
+  selector: 'aio-events',
+  templateUrl: 'events.component.html'
+})
+export class EventComponent implements OnInit {
+
+  prevEvents: Event[];
+  upcomingEvents: Event[];
+
+  constructor( private eventsService: EventsService ) { }
+
+  ngOnInit() {
+    this.eventsService.events.subscribe(( events: Event[] ) => {
+      this.prevEvents = events.filter( event => {
+        return new Date(event.endDate).valueOf() < Date.now();
+      });
+
+      this.upcomingEvents = events.filter( event => {
+        return new Date(event.endDate).valueOf() > Date.now();
```

Comment:
So, for a very brief moment in time an event won't be listed in any table :grin:

```suggestion
        return new Date(event.endDate).valueOf() >= Date.now();
```

Actually, I think the logic is not correct for determining where to show an event. When using `new Date('YYYY-MM-DD')` it implies `00:00` as time. I think we should show the event as "upcoming" during the last day and only move it to "past events" after that.

So, insted of comparing against `Date.now()`, we should be comparing against `Date.now() - (/* One day */ 24 * 60 * 60 * 1000)`.